libraries/common/subscriptions/helpers/buildRegisterUrl.js

```diff
@@ -0,0 +1,32 @@
+import queryString from 'query-string';
+import {
+  CHECKOUT_PATH,
+  INDEX_PATH,
+} from '../../constants/RoutePaths';
+
+/**
+ * Generates an url for registrations via the webcheckout. It injects additional parameters
+ * for redirections after a successful registration or checkout.
+ * @param {string} url The pure register url.
+ * @param {string} [redirect=''] An url which is supposed to be opened after registration.
+ * @return {string}
+ */
+const getRegisterUrl = (url, redirect = '') => {
+  const parsed = queryString.parseUrl(url);
+
+  // Generate the final redirect url. Checkout redirects are mapped to the start page.
+  const redirectTo = (redirect && !redirect.startsWith(CHECKOUT_PATH)) ? redirect : INDEX_PATH;
+
+  // Build the callback data.
+  parsed.query.sgcloud_callback_data = JSON.stringify({ redirectTo });
+
+  // Explicitly check if we are about to be redirected to the checkout.
+  if (redirect && redirect.startsWith(CHECKOUT_PATH)) {
```

Comment:
The first check doesn't make sense here. If the `redirect` param is maybe not set, it falls back to an empty string. Please don't give it a default, if you check it either way. Or don't check it again. One of both is redundant.

Reply:
done